### ISSUE TITLE
Reproducible build failures should upload artifacts

### ISSUE THREAD

**Issue by jsoref** [apache/logging-log4j2#1318]
https://github.com/check-spelling/logging-log4j2/actions/runs/4309470335/jobs/7516899702#step:5:363

```
[INFO] Minimal buildinfo generated from downloaded artifacts: /home/runner/work/logging-log4j2/logging-log4j2/log4j-api/target/reference/log4j-api-3.0.0-SNAPSHOT.buildinfo
Error:  sha512 mismatch log4j-api-3.0.0-SNAPSHOT.jar: investigate with diffoscope log4j-api/target/reference/log4j-api-3.0.0-SNAPSHOT.jar log4j-api/target/log4j-api-3.0.0-SNAPSHOT.jar
Error:  Reproducible Build output summary: 3 files ok, 1 different
Error:  see diff log4j-api/target/reference/log4j-api-3.0.0-SNAPSHOT.buildinfo log4j-api/target/log4j-api-3.0.0-SNAPSHOT.buildinfo
Error:  see also https://maven.apache.org/guides/mini/guide-reproducible-builds.html
[INFO] Reproducible Build output comparison saved to /home/runner/work/logging-log4j2/logging-log4j2/log4j-api/target/log4j-api-3.0.0-SNAPSHOT.buildcompare
[INFO] Aggregate buildcompare copied to /home/runner/work/logging-log4j2/logging-log4j2/target/log4j-3.0.0-SNAPSHOT.buildcompare
```

This failure should trigger an `actions/upload-artifact` step that uploads all the relevant files so that contributors (especially drive-by) can see the things that it doesn't like.

**Comment by jsoref** [comment]
Here's a "simpler" run: https://github.com/check-spelling/logging-log4j2/actions/runs/4309681682/jobs/7517347157

which is composed of a single commit:
https://github.com/apache/logging-log4j2/commit/6ce75ef95163b067348973b3d731b8699ba1eb5a

which afaict, is *only* applying changes as directed by spotless.

**Comment by ppkarwasz** [comment]
@jsoref,

Sure, these failures should be debuggable. Can you provide a PR?

**Comment by jsoref** [comment]
https://github.com/check-spelling/logging-log4j2/pull/4

**Comment by jsoref** [comment]
Oh, sorry, you mean a PR to actually implement this?

_maybe_. Remember, I'm very much trying to do a drive by contribution and tripping on a dozen problems.

If I knew precise paths for the output objects, I could probably provide the PR for the workflow changes... I'm fairly comfortable w/ github action workflow definitions. I'm not remotely comfortable w/ mvn.

**Comment by ppkarwasz** [comment]
Yes, a PR solving the problem would be nice.

The artifacts are probably in `<module_name>/target/*.jar` and `<module_name>/target/reference/*.jar`. I am not very versed in Github Action definitions.

BTW: I was able to identify the cause of the failure.

**Comment by jsoref** [comment]
Here's a run w/ an artifact: https://github.com/check-spelling/logging-log4j2/actions/runs/4318380619

**Comment by jsoref** [comment]
And here's what the build compare says, it doesn't mean anything to me:
```sh
% for a in $(find . -name '*buildcompare'); do echo $a; cat $a; echo; done
./target/log4j-3.0.0-SNAPSHOT.buildcompare
version=3.0.0-SNAPSHOT
ok=3
ko=1
okFiles="log4j-api-3.0.0-SNAPSHOT.pom log4j-api-3.0.0-SNAPSHOT-sources.jar log4j-api-3.0.0-SNAPSHOT-test-sources.jar"
koFiles="log4j-api-3.0.0-SNAPSHOT.jar"
reference_java_version="11 (from MANIFEST.MF Build-Jdk-Spec)"
reference_os_name="Unix (from pom.properties newline)"
# diffoscope log4j-api/target/reference/log4j-api-3.0.0-SNAPSHOT.jar log4j-api/target/log4j-api-3.0.0-SNAPSHOT.jar

./log4j-api/target/log4j-api-3.0.0-SNAPSHOT.buildcompare
version=3.0.0-SNAPSHOT
ok=3
ko=1
okFiles="log4j-api-3.0.0-SNAPSHOT.pom log4j-api-3.0.0-SNAPSHOT-sources.jar log4j-api-3.0.0-SNAPSHOT-test-sources.jar"
koFiles="log4j-api-3.0.0-SNAPSHOT.jar"
reference_java_version="11 (from MANIFEST.MF Build-Jdk-Spec)"
reference_os_name="Unix (from pom.properties newline)"
# diffoscope log4j-api/target/reference/log4j-api-3.0.0-SNAPSHOT.jar log4j-api/target/log4j-api-3.0.0-SNAPSHOT.jar
```

**Comment by vy** [comment]
@jsoref, have you tried the `diffoscope log4j-api/target/reference/log4j-api-3.0.0-SNAPSHOT.jar log4j-api/target/log4j-api-3.0.0-SNAPSHOT.jar` hinted in the failure? (You can also unzip the JARs and compare them with `diff`.)

**Comment by jsoref** [comment]
@vy: that's out of scope for this PR.

I'm not trying to boil oceans. I will say that `diffoscope ...` fails because I don't have it installed, and as a drive-by-contributor, being asked to have hundreds of random programs installed on my computer and understand how they work is really asking quite a bit. Even being asked to file the bug for this and this PR is a bit much. I'm doing each item as a donation to a project I barely use.

If you want to enhance things further after #1321 merges, you're welcome to do so, and certainly that output would be a significant improvement over the current behavior or even the incremental improvement provided by the PR.

```diff
diffoscope log4j-api/target/reference/log4j-api-3.0.0-SNAPSHOT.jar log4j-api/target/log4j-api-3.0.0-SNAPSHOT.jar
--- log4j-api/target/reference/log4j-api-3.0.0-SNAPSHOT.jar
+++ log4j-api/target/log4j-api-3.0.0-SNAPSHOT.jar
├── zipinfo {}
```
```diff
@@ -211,9 +211,9 @@
 -rw----     2.0 fat      386 bX defN 79-Nov-30 04:08 org/apache/logging/log4j/util/TriConsumer.class
 -rw----     2.0 fat     1144 bX defN 79-Nov-30 04:08 org/apache/logging/log4j/util/Unbox$State.class
 -rw----     2.0 fat     1572 bX defN 79-Nov-30 04:08 org/apache/logging/log4j/util/Unbox$WebSafeState.class
 -rw----     2.0 fat     4419 bX defN 79-Nov-30 04:08 org/apache/logging/log4j/util/Unbox.class
 -rw----     2.0 fat      135 bX defN 79-Nov-30 04:08 org/apache/logging/log4j/util/package-info.class
 -rw----     2.0 fat     4708 bX defN 79-Nov-30 04:08 META-INF/maven/org.apache.logging.log4j/log4j-api/pom.xml
 -rw----     2.0 fat       77 bX defN 79-Nov-30 04:08 META-INF/maven/org.apache.logging.log4j/log4j-api/pom.properties
--rw----     2.0 fat      980 bl defN 23-Mar-02 22:09 module-info.class
+-rw----     2.0 fat      980 bl defN 23-Mar-02 22:13 module-info.class
 217 files, 775553 bytes uncompressed, 275425 bytes compressed:  64.5%
```

I think that @ppkarwasz has a fix for the output.
